mmv1/products/gkeonprem/VmwareCluster.yaml

```diff
@@ -0,0 +1,662 @@
+      The DNS name of VMware User Cluster's API server.
+    output: true
+  - !ruby/object:Api::Type::Boolean
+    name: "reconciling"
+    description: |
+      If set, there are currently changes in flight to the VMware User Cluster.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: "createTime"
+    description: |
+      The time at which VMware User Cluster was created.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: "updateTime"
+    description: |
+      The time at which VMware User Cluster was last updated.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: "deleteTime"
+    description: |
+      The time at which VMware User Cluster was deleted.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: "localName"
+    description: |
+      The object name of the VMware OnPremUserCluster custom resource on the
+      associated admin cluster. This field is used to support conflicting
+      names when enrolling existing clusters to the API. When used as a part of
+      cluster enrollment, this field will differ from the ID in the resource
+      name. For new clusters, this field will match the user provided cluster ID
+      and be visible in the last component of the resource name. It is not
+      modifiable.
+
+      All users should use this name to access their cluster using gkectl or
+      kubectl and should expect to see the local name when viewing admin
+      cluster controller logs.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: "etag"
+    description: |
+      This checksum is computed by the server based on the value of other
+      fields, and may be sent on update and delete requests to ensure the
+      client has an up-to-date value before proceeding.
+      Allows clients to perform consistent read-modify-writes
+      through optimistic concurrency control.
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: "annotations"
+    description: |
+      Annotations on the VMware User Cluster.
+      This field has the same restrictions as Kubernetes annotations.
+      The total size of all keys and values combined is limited to 256k.
+      Key can have 2 segments: prefix (optional) and name (required),
+      separated by a slash (/).
+      Prefix must be a DNS subdomain.
+      Name must be 63 characters or less, begin and end with alphanumerics,
+      with dashes (-), underscores (_), dots (.), and alphanumerics between.
+    default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: "controlPlaneNode"
+    description: |
+      VMware User Cluster control plane nodes must have either 1 or 3 replicas.
+    required: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: cpus
+        description: |
+          The number of CPUs for each admin cluster node that serve as control planes
+          for this VMware User Cluster. (default: 4 CPUs)
+        default_from_api: true
```

Comment:
If these values have fixed defaults, should we consider setting clientside defaults? That'll create clearer plans, and safeguard against the zero value of the type being misunderstood. Terraform types are primitive, and `default_from_api` adds a behaviour flag that considers the zero value to mean "output only", even when explicitly set.

Reply:
Added default_value for each field. Thanks for the guidance.